File: Musher/src/song_test.cpp
#include <iostream>
#include <string>

#include "song.h"
#include "functional_test.h"


int main()
{
    std::cout << "================================================================" << std::endl;
    std::cout << "This is the c++ wrapper to test the Song shared library"          << std::endl;
    std::cout << "================================================================" << std::endl;

    Song first_song("My First Song", 4.5);
    Song second_song("My Second Song", 2.3);

    std::cout << first_song.GetName() << " " << first_song.GetLength() << std::endl;
    std::cout << second_song.GetName() << " " << second_song.GetLength() << std::endl;

    first_song.PrintMessage();
    second_song.PrintMessage();

    PrintFunctionalMessage("This is a message for a functional program");

    return 0;
}

File: poc/c++_extensions/song/wrapper/song_test.py
from song import *


def main():
    print("================================================================")
    print("This is the python wrapper to test the Song shared library")
    print("================================================================")

    first_song = song("My First Song", 6.2)
    second_song = song("My Second Song", 12)

    print(first_song.GetName() + " " + str(first_song.GetLength()))
    print(second_song.GetName() + " " + str(second_song.GetLength()))

    PrintFunctionalMessage("This is a functional message")


if __name__ == "__main__":
    main()
